tests/integrationv2/test_sslyze.py

```diff
@@ -0,0 +1,289 @@
+        else:
+            assert self.scan_result.tls_ticket_resumption_result == sslyze.TlsResumptionSupportEnum.FULLY_SUPPORTED
+
+
+class CrimeVerifier(ScanVerifier):
+    def assert_scan_success(self):
+        assert self.scan_result.supports_compression is False
+
+
+class EarlyDataVerifier(ScanVerifier):
+    def assert_scan_success(self):
+        if self.protocol == Protocols.TLS13:
+            assert self.scan_result.supports_early_data is True
+        else:
+            assert self.scan_result.supports_early_data is False
+
+
+class DowngradePreventionVerifier(ScanVerifier):
+    def assert_scan_success(self):
+        assert self.scan_result.supports_fallback_scsv is True
+
+
+class HeartbleedVerifier(ScanVerifier):
+    def assert_scan_success(self):
+        assert self.scan_result.is_vulnerable_to_heartbleed is False
+
+
+class CCSInjectionVerifier(ScanVerifier):
+    def assert_scan_success(self):
+        assert self.scan_result.is_vulnerable_to_ccs_injection is False
+
+
+class InsecureRenegotiationVerifier(ScanVerifier):
+    def assert_scan_success(self):
+        assert self.scan_result.is_vulnerable_to_client_renegotiation_dos is False
+
+
+def validate_scan_result(scan_attempt, protocol, certificate=None):
+    assert_scan_attempt_completed(scan_attempt)
+    scan_result = scan_attempt.result
+
+    verifier_cls = {
+        sslyze.CipherSuitesScanResult:              CipherSuitesVerifier,
+        sslyze.SupportedEllipticCurvesScanResult:   EllipticCurveVerifier,
+        sslyze.RobotScanResult:                     RobotVerifier,
+        sslyze.SessionResumptionSupportScanResult:  SessionResumptionVerifier,
+        sslyze.CompressionScanResult:               CrimeVerifier,
+        sslyze.EarlyDataScanResult:                 EarlyDataVerifier,
+        sslyze.FallbackScsvScanResult:              DowngradePreventionVerifier,
+        sslyze.HeartbleedScanResult:                HeartbleedVerifier,
+        sslyze.OpenSslCcsInjectionScanResult:       CCSInjectionVerifier,
+        sslyze.SessionRenegotiationScanResult:      InsecureRenegotiationVerifier
+    }.get(type(scan_result))
+
+    assert verifier_cls is not None, f"unexpected scan: {scan_attempt}"
+
+    verifier = verifier_cls(scan_result, protocol, certificate)
+    verifier.assert_scan_success()
+
+
+def get_scan_attempts(scan_results):
+    # scan_results (sslyze.AllScanCommandsAttempts) is an object containing parameters mapped to scan attempts. convert
+    # this to a list containing just scan attempts, and then filter out tests that were not scheduled.
+    scan_attribute_names = [attr_name for attr_name in dir(scan_results) if not attr_name.startswith("__")]
+    scan_attempts = [getattr(scan_results, attr_name) for attr_name in scan_attribute_names]
+    scan_attempts = [
+        scan_attempt for scan_attempt in scan_attempts
+        if scan_attempt.status != sslyze.ScanCommandAttemptStatusEnum.NOT_SCHEDULED
```

Comment:
This is a nit, but I generally avoid double negatives if possible.
```suggestion
        if scan_attempt.status == sslyze.ScanCommandAttemptStatusEnum.SCHEDULED
```

Reply:
Ah, right. However, in this case I don't think it's possible because there is a ScanCommandAttemptStatusEnum.Error value that I actually want to let through so it fails the scan attempt assertion. Otherwise, it would just get filtered out and then the test would be skipped with no failure.